src/course-outline/configure-modal/ConfigureModal.jsx

```diff
@@ -16,33 +16,123 @@ import { getCurrentItem } from '../data/selectors';
 import messages from './messages';
 import BasicTab from './BasicTab';
 import VisibilityTab from './VisibilityTab';
+import AdvancedTab from './AdvancedTab';
 
 const ConfigureModal = ({
   isOpen,
   onClose,
   onConfigureSubmit,
 }) => {
   const intl = useIntl();
-  const { displayName, start: sectionStartDate, visibilityState } = useSelector(getCurrentItem);
+  const {
+    displayName,
+    start: sectionStartDate,
+    visibilityState,
+    due,
+    isTimeLimited,
+    defaultTimeLimitMinutes,
+    hideAfterDue,
+    showCorrectness,
+    courseGraders,
+    category,
+    format,
+  } = useSelector(getCurrentItem);
   const [releaseDate, setReleaseDate] = useState(sectionStartDate);
   const [isVisibleToStaffOnly, setIsVisibleToStaffOnly] = useState(visibilityState === VisibilityTypes.STAFF_ONLY);
   const [saveButtonDisabled, setSaveButtonDisabled] = useState(true);
+  const [graderType, setGraderType] = useState(format == null ? 'Not Graded' : format);
+  const [dueDateState, setDueDateState] = useState('');
+  const [isTimeLimitedState, setIsTimeLimitedState] = useState(false);
+  const [defaultTimeLimitMin, setDefaultTimeLimitMin] = useState(30);
+  const [hideAfterDueState, setHideAfterDueState] = useState(false);
+  const [showCorrectnessState, setShowCorrectnessState] = useState(false);
+  const [isSubsection, setIsSubsection] = useState(category === 'sequential');
 
   useEffect(() => {
     setReleaseDate(sectionStartDate);
   }, [sectionStartDate]);
 
+  useEffect(() => {
+    setIsSubsection(category === 'sequential');
+  }, [category]);
+
+  useEffect(() => {
+    setGraderType(format == null ? 'Not Graded' : format);
+  }, [format]);
+
+  useEffect(() => {
+    setDueDateState(due);
+  }, [due]);
+
+  useEffect(() => {
+    setIsTimeLimitedState(isTimeLimited);
+  }, [isTimeLimited]);
+
+  useEffect(() => {
+    setDefaultTimeLimitMin(defaultTimeLimitMinutes);
+  }, [defaultTimeLimitMinutes]);
+
+  useEffect(() => {
+    setHideAfterDueState(hideAfterDue);
+  }, [hideAfterDue]);
+
+  useEffect(() => {
+    setShowCorrectnessState(showCorrectness);
+  }, [showCorrectness]);
+
```

Comment:
This is necessary because the initial values are given by the redux state which is asynchronous. In order to synchronize both redux state and react states this use effects are necessary. Please see https://stackoverflow.com/questions/70292062/how-can-i-set-initial-value-for-usestate-from-redux-state-using-selector-in-reac as reference

Reply:
Makes sense, but this way of handling forms is not good. We should have used `Formik` like in [src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx](https://github.com/open-craft/frontend-app-course-authoring/blob/3f676493ef57c00002cb604cef565603b8d98c85/src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx#L176-L189).

For now we can leave it as is, but we'll need to refactor this at some point. Can you add a todo comment somewhere in this component.